component/stream/stream.go

```diff
@@ -289,10 +266,10 @@ func (st *Stream) copyCachedBlock(fileKey string, handle *handlemap.Handle, offs
 		// round all offsets to the specific blocksize offsets
 		cachedBlockStartIndex := (offset - (offset % st.streamCache.blockSize))
 		// Lock on requested block and fileName to ensure it is not being rerequested or manipulated
-		block, exists, err := st.fetchBlock(fileKey, handle, cachedBlockStartIndex)
+		block, exists, err := st.fetchBlock(handle, cachedBlockStartIndex)
```

Comment:
When write stream comes this offset based calculation of figuring out the block will not work as different blocks can be of different sizes. We need to reorg this code where block search is based on whether its read-only or read-write streaming.

Reply:
yea when putting the pr for write stream I will create a base class where read and write stream inherit from - read stream will have a different block definition from write stream. This will not be in this PR.